executor/bind.go

```diff
@@ -136,6 +136,10 @@ func (e *SQLBindExec) createSQLBind() error {
 	// is necessary to avoid 'create binding' been recorded as 'explain'.
 	saveStmtCtx := e.Ctx().GetSessionVars().StmtCtx
 	defer func() {
+		// But we need to restore the SET_VAR's setting.
+		for name, val := range e.Ctx().GetSessionVars().StmtCtx.SetVarHintRestore {
+			saveStmtCtx.AddSetVarHintRestore(name, val)
+		}
```

Comment:
I suggest we add some comments on why this is needed, like where is the `SetVarHintRestore` set.
It's really not easy to realize.

Reply:
And looks like we are solving two problems in this PR. I think it's better to clarify it in the issue or PR description.